File: professional/ele-32/lab3/binary/bsc_canal.cpp
#include "bsc_canal.h"

BscCanal::BscCanal(double p, Random *r) : p(p), r(r) {
}

int BscCanal::transmit(int x) {
	if (this->errorHappened()) {
		return 1 - x;
	}
	return x;
}

bool BscCanal::errorHappened() {
	double u = this->r->uniformZeroOne();
	return (u <= this->p);
}

File: professional/third-semester/ele-32/lab2/bscCanal.m
function y = bscCanal(x,p)

    if errorOccurs(p)        
        y = 1 - x;
    else
        y = x;
    end
    
end

function e = errorOccurs(p)
    r = random('Uniform', 0, 1);
    
    e = r<=p;
end